### DISCUSSION TITLE
How can I know details about struct lldpd_port?

### DISCUSSION THREAD

**Discussion by watsheng** [lldpd/lldpd#562] [Q&A] [answered]
Hello!
I git clone the lldpd source code to learn how to implement lldp. However I can't view struct lldpd_port and some other structs. I want to know details. How can I view these structs? If I want to develop my own simple lldpd and use these kind of structs, which header files or which libs I need to include? Thanks for your graceful help!
Best Wishes,
Watson.

**Comment by vincentbernat** [comment, marked as answer]
It's defined in `src/lldpd-structs.h`. Not sure the code is a good example on how to start a project from scratch. Better look at a simpler implementation, like the one in systemd.

**Reply by watsheng** [reply]
Thanks for your quickly response! Yes, I see, your lldpd project is more comprehensive and compatible with other discovery protocols. Could you please tell me more infos about "the simpler implementation in systemd"? I'm not quite get your words.
